### PR TITLE
L2 get vendor request

### PR DESCRIPTION
**Pull request by simen-rekkedal**
## Related Issue(s)
Authentication API for opprette forespørsel [- #586](https://github.com/Altinn/altinn-authentication/issues/586)

## Verification
- [x] **Your** code builds clean without any errors or warnings
- [ ] Manual testing done (required)
- [ ] Relevant automated test added (if you find this hard, leave it and we'll help out)
- [ ] All tests run green

### CODE REVIEW

File: bff/src/Altinn.Authentication.UI/Altinn.Authentication.UI.Integration/SystemUsers/RequestClient.cs [hunk 1]
```diff
@@ -0,0 +1,46 @@
+﻿using Altinn.Authentication.UI.Core.Authentication;
+using Altinn.Authentication.UI.Core.SystemUsers;
+using Altinn.Authentication.UI.Integration.AccessToken;
+using Altinn.Authentication.UI.Integration.Configuration;
+using Altinn.Authorization.ProblemDetails;
+using Microsoft.AspNetCore.Http;
+using Microsoft.Extensions.Options;
+using Altinn.Authentication.UI.Core.Extensions;
+using System.Text.Json;
```
Comment (acn-dgopa) [inline bff/src/Altinn.Authentication.UI/Altinn.Authentication.UI.Integration/SystemUsers/RequestClient.cs:9]: i dont see any serializing casing options but i see you have defined camelcasing on the properties

File: bff/src/Altinn.Authentication.UI/Altinn.Authentication.UI.Core/SystemUsers/IRequestClient.cs [hunk 2]
```diff
@@ -0,0 +1,8 @@
+﻿using Altinn.Authorization.ProblemDetails;
+
+namespace Altinn.Authentication.UI.Core.SystemUsers;
+
+public interface IRequestClient
+{    
+    Task<Result<VendorRequest>> GetVendorRequest(int partyId, Guid requestId);
```
Comment (mgunnerud) [inline bff/src/Altinn.Authentication.UI/Altinn.Authentication.UI.Core/SystemUsers/IRequestClient.cs:7]: Should `cancellationToken` be added to parameters?

**Review by mgunnerud — CHANGES_REQUESTED** [review]

File: bff/src/Altinn.Authentication.UI/Altinn.Authentication.UI.Integration/SystemUsers/RequestClient.cs (hunk 1 above)
Comment (simen-rekkedal) [reply bff/src/Altinn.Authentication.UI/Altinn.Authentication.UI.Integration/SystemUsers/RequestClient.cs:9]: ok fixed

Comment (simen-rekkedal) [reply bff/src/Altinn.Authentication.UI/Altinn.Authentication.UI.Integration/SystemUsers/RequestClient.cs:9]: ok, fixed

File: bff/src/Altinn.Authentication.UI/Altinn.Authentication.UI.Core/SystemUsers/IRequestClient.cs (hunk 2 above)
Comment (simen-rekkedal) [reply bff/src/Altinn.Authentication.UI/Altinn.Authentication.UI.Core/SystemUsers/IRequestClient.cs]: ok fixed

File: bff/src/Altinn.Authentication.UI/Altinn.Authentication.UI/Controllers/RequestController.cs [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+{       
+    /// <summary>
+    /// Gets a VendorRequest by Id
+    /// </summary>
+    /// <returns></returns>
+    [Authorize]
+    [HttpGet("{requestId}")]
+    public async Task<ActionResult> GetRequestByPartyIdAndRequestId(Guid requestId)
+    {
+        int partyId = AuthenticationHelper.GetRepresentingPartyId(HttpContext);
```
Comment (TheTechArch) [inline bff/src/Altinn.Authentication.UI/Altinn.Authentication.UI/Controllers/RequestController.cs:28]: Hvordan funker dette? Ser da ikke party i route?

Comment (TheTechArch) [reply bff/src/Altinn.Authentication.UI/Altinn.Authentication.UI/Controllers/RequestController.cs:28]: Eller. dette er bff. og her har dere vel noe cookie opplegg

Comment (simen-rekkedal) [reply bff/src/Altinn.Authentication.UI/Altinn.Authentication.UI/Controllers/RequestController.cs:28]: Ja, henter ut party fra bearer token

**Review by mgunnerud — APPROVED** [review]

**Pull request merged**